accepted/single-file/extract.md

```diff
@@ -73,62 +63,48 @@ The desirable characteristics of the extraction directory are:
-         * On startup:
-            * If the symbolic link exists and isn't stale (points to a directory owned by the user, with correct permissions (`0700`), etc.), that's what it is used;
-            * If the symbolic link does not exist (or exists and is stale), it is removed, a new directory is created with `mkdtemp()`, and the link is re-created.
+  * If not, defaults to: 
+    * Windows: `%TEMP%\.net`
+    * Unix:
+      * `${TMPDIR}/.net/${UID}` if `${TMPDIR}` is set; otherwise,
+      * `/var/tmp/.net/${UID}` if `/var/tmp` exists, and is writable; otherwise,
+      * `/tmp/.net/${UID}` if `/tmp` exists, and is writable; otherwise fail.
 
 * `<app>` is the name of the single-exe binary
+* `<bundle-id>` is the unique [Bundle-identifier](bundler.md#bundle identifier). 
 
-* `<bundle-id>` is a unique Bundle-identifier. 
-
-  Each single-file bundle contains a path-compatible cryptographically strong identifier embedded within it. This identifier is used as part of the extraction path in order to ensure that files from multiple versions of an app do not interfere with each other.
-
-#### Cleanup
+#### Unix: An Alternate Proposal 
 
-The cleanup of extracted files in the install-location will be manual in this version. 
+On Unix-like systems, where multiple users may use a single system, an approach that removes the possibility of name collisions and other users creating files to prevent an application to start (by a malicious user creating a predictable directory name) is used instead:
 
-We can consider providing helper scripts to cleanup the extracted files for an executable. However, future versions are expected to spill fewer artifacts to disk, making the cleanup commands a lower priority feature.
+* The directory to extract the bundle is created with `mkdtemp()`, using the `$TMPDIR` environment variable (if set), `/var/tmp/` (if exists, because it survives reboots), falling back to `/tmp` (does not survive reboots, it's often a ramdisk) in the template (e.g. `/var/tmp/dotnet-<app>-XXXXXX`);
+  * To facilitate reuse of extraction, the extraction directory must be predictable. To achieve this, a symbolic link to the directory created by `mkdtemp()` is created in a predictable location:
+    * If `${XDG_CACHE_HOME}` is set, the symlink is created under `${XDG_CACHE_HOME}/.cache/dotnet/<app>/<bundle-id>` (See [the XDG spec for information](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)); otherwise,
+    * Otherwise, the symlink is created under `~/.cache/dotnet/<app>/<bundle-id>`
+  * In order to check for reuse on startup:
+    * If the symbolic link exists and isn't stale (points to a directory owned by the user, with correct permissions (`0700`), etc.), that's what it is used;
+    * If the symbolic link does not exist (or exists and is stale), it is removed, a new directory is created with `mkdtemp()`, and the link is re-created.
 
```

Comment:
@lpereira worked on prototyping another alternate mechanism for extracting files on Linux (when necessary) using [FUSE](http://man7.org/linux/man-pages/man4/fuse.4.html) -- to extract to an in-memory file system. @lpereira  can add additional details.